vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -402,12 +441,40 @@ void AdsImpl::NotificationEventDismissed(
 void AdsImpl::NotificationEventTimedOut(
     const std::string& id,
     const NotificationInfo& notification) {
-  notifications_->Remove(id);
+  notifications_->Remove(id, false);
 
   GenerateAdReportingNotificationResultEvent(notification,
       NotificationResultInfoResultType::TIMEOUT);
 }
 
+bool AdsImpl::IsDoNotDisturb() const {
+  if (!IsAndroid()) {
```

Comment:
I think this should be a feature check, not a platform check. `IsDoNotDisturbedEnabled` that only returns true for android

Reply:
I will add a ticket, and if ok fix this as a wider refactor for any of these checks

Reply:
@bridiver https://github.com/brave/brave-browser/issues/6229